system/routes_test.go

```diff
@@ -27,22 +44,187 @@ func TestUsers(t *testing.T) {
+		resp := recorder.Result()
+
+		fmt.Println(">>> (request)")
+		fmt.Println(request(req))
+		fmt.Println("----")
+		fmt.Println("<<< (response)")
+		fmt.Println(response(resp))
+
+		jr, err := decodeJson(resp.Body)
+		assert(t, err == nil, "Error decoding response body: %+v", err)
+		assert(t, jr.Error.Message == "missing form body", "Expected error 'missing form body' got: %+v", jr.Error.Message)
+	}
+
+	// Send "Login" request with missing user.
+	{
+		json := []byte(`{"username":"test123","password":"test123"}`)
+
+		req, err := http.NewRequest("POST", "http://localhost/auth/login", bytes.NewBuffer(json))
+		req.Header.Set("Content-Type", "application/json")
+
+		assert(t, err == nil, "Error creating request: %+v", err)
+
+		recorder := httptest.NewRecorder()
+		Routes().ServeHTTP(recorder, req)
+		resp := recorder.Result()
+
+		fmt.Println(">>> (request)")
+		fmt.Println(request(req))
+		fmt.Println("----")
+		fmt.Println("<<< (response)")
+		fmt.Println(response(resp))
+
+		jr, err := decodeJson(resp.Body)
+		assert(t, err == nil, "Error decoding response body: %+v", err)
+		assert(t, jr.Error.Message == "crust.auth.repository.UserNotFound", "Expected error 'crust.auth.repository.UserNotFound' got: %+v", jr.Error.Message)
+	}
+
+	// Create user.
+	user := &systemTypes.User{
+		ID:       1337,
+		Username: "johndoe",
+	}
+	{
+		err := user.GeneratePassword("johndoe123")
+		assert(t, err == nil, "Error generating password: %+v", err)
+	}
+	{
+		userAPI := systemRepository.User(context.Background(), nil)
+		_, err := userAPI.Create(user)
+		assert(t, err == nil, "Error when inserting user: %+v", err)
+	}
+
+	// Send "Login" request with existing user.
+	{
+		form := url.Values{}
+		form.Add("username", "johndoe")
+		form.Add("password", "johndoe123")
+
+		req, err := http.NewRequest("POST", "http://localhost/auth/login", strings.NewReader(form.Encode()))
+		req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+
+		assert(t, err == nil, "Error creating request: %+v", err)
+
+		recorder := httptest.NewRecorder()
+		Routes().ServeHTTP(recorder, req)
+		resp := recorder.Result()
+
+		fmt.Println(">>> (request)")
+		fmt.Println(request(req))
+		fmt.Println("----")
+		fmt.Println("<<< (response)")
+		fmt.Println(response(resp))
+
+		c := resp.Cookies()
+		assert(t, len(c) == 1, "Expected 1 cookie value, got: %+v", len(c))
+		assert(t, c[0].Value != "", "Expected non empty jwt token, got: %+v", c[0].Value)
+
+		type jsonResponse struct {
+			Response struct {
+				UserID   string `json:"userID"`
+				Username string `json:"username"`
+			} `json:"response"`
+		}
+
+		var jr jsonResponse
+		err = json.NewDecoder(resp.Body).Decode(&jr)
+		assert(t, err == nil, "Error decoding response body: %+v", err)
+		assert(t, jr.Response.UserID != "0", "Expected userID not to be 0, got: %+v", jr.Response.UserID)
+		assert(t, jr.Response.Username == "johndoe", "Expected username 'johndoe', got: %+v", jr.Response.Username)
+	}
+
```

Comment:
Test `/auth/check` here to see if login returns valid token.

Reply:
Changed.